builder/dockerfile/internals.go

```diff
@@ -435,14 +435,18 @@ func (b *Builder) processImageFrom(img builder.Image) error {
 		}
 
 		total := len(ast.Children)
-		for i, n := range ast.Children {
+		for _, n := range ast.Children {
 			switch strings.ToUpper(n.Value) {
 			case "ONBUILD":
 				return fmt.Errorf("Chaining ONBUILD via `ONBUILD ONBUILD` isn't allowed")
 			case "MAINTAINER", "FROM":
 				return fmt.Errorf("%s isn't allowed as an ONBUILD trigger", n.Value)
 			}
-
+			if err := b.evaluateDispatch(n); err != nil {
```

Comment:
if I'm following, I'd prefer to undo this split here and just do the syntax check in the `evaluateDispatch()` func above.